src/tests/roc_fec/target_openfec/test_writer_reader.cpp

```diff
@@ -1653,5 +1653,129 @@ TEST(writer_reader, error_reader_decode_packet) {
     }
 }
 
+TEST(writer_reader, writer_oversized_block) {
+    for (size_t n_scheme = 0; n_scheme < Test_n_fec_schemes; ++n_scheme) {
+        config.scheme = Test_fec_schemes[n_scheme];
+
+        OFEncoder encoder(config, FECPayloadSize, allocator);
+        OFDecoder decoder(config, FECPayloadSize, buffer_pool, allocator);
+
+        CHECK(encoder.valid());
+        CHECK(decoder.valid());
+
+        CHECK(decoder.max_block_length() == encoder.max_block_length());
+        CHECK(NumSourcePackets + NumRepairPackets < encoder.max_block_length());
+
+        PacketDispatcher dispatcher(NumSourcePackets, NumRepairPackets);
+
+        Writer writer(config, FECPayloadSize, encoder, dispatcher, source_composer(),
+                      repair_composer(), packet_pool, buffer_pool, allocator);
+
+        Reader reader(config, decoder, dispatcher.source_reader(),
+                      dispatcher.repair_reader(), rtp_parser, packet_pool, allocator);
+
+        CHECK(writer.valid());
+        CHECK(reader.valid());
+
+        // try to resize writer with an invalid value
+        CHECK(!writer.resize(encoder.max_block_length() * 2));
```

Comment:
I suggest to try `encoder.max_block_length() + 1` instead to catch off-by-one errors.

Reply:
Done